vocode/streaming/agent/state_agent.py

```diff
@@ -453,6 +444,8 @@ def __init__(
         self,
         agent_config: CommandAgentConfig,
         logger: Optional[logging.Logger] = None,
+        *,
+        events_manager: Optional[EventsManager] = None,
```

Comment:
from an architecture perspective, I don't love coupling Agents with EventsManager. There's already lots of annoying coupling in Vocode, this one is adding a surprising amount of code complexity and I think the agent's runtime is semantically distinct from the agent itself

I think it would be easy+slightly cleaner to pass a callback. Will add example in a top level comment

Reply:
I did because I wanted to send an Event everytime I did generated a completion

Reply:
The coupling is not that tight since it's optional, it is just explicit, you are already implicitly coupling because you call a bunch of functions and attributes which are only on the state agent in the streaming conversation, but you just ignore the type errors. My one just allows it take an event manager and broadcast some events since I need it to be broadcasted from inside the class when there are state changes inside the class. If you mean. The other option is to not have it bind the event manager inside the streaming conversation, but I already looked at that but the problem with that is that if the event manager is not defined then the streaming conversation has a default event manager, and I thought it was more complex having it accidentally be two event managers.